File: codegen/results/v001/gemma4_26b_think/20260407_163120/go/004_json_filter/solution.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"sort"
)

// Record represents the structure of the objects in the JSON array.
type Record struct {
	Name   string  `json:"name"`
	Age    int     `json:"age"`
	Active bool    `json:"active"`
	Score  float64 `json:"score"`
}

func main() {
	// Path to the input file
	filePath := "input/data.json"

	// Read the file contents
	data, err := os.ReadFile(filePath)
	if err != nil {
		// If the file cannot be read, we exit without printing to stdout 
		// to adhere to the "Do not output anything else" constraint.
		os.Exit(1)
	}

	// Parse the JSON array
	var records []Record
	if err := json.Unmarshal(data, &records); err != nil {
		os.Exit(1)
	}

	// Filter records: active must be true and age must be 30 or older
	var filtered []Record
	for _, r := range records {
		if r.Active && r.Age >= 30 {
			filtered = append(filtered, r)
		}
	}

	// Sort the filtered records by name in ascending order
	sort.Slice(filtered, func(i, j int) bool {
		return filtered[i].Name < filtered[j].Name
	})

	// Encode the resulting slice back to JSON
	// We use MarshalIndent for clean, readable output, but standard Marshal works too.
	output, err := json.MarshalIndent(filtered, "", "  ")
	if err != nil {
		os.Exit(1)
	}

	// Write the JSON array to stdout
	fmt.Println(string(output))
}

File: codegen/results/v001/gemma4_26b_think/20260407_163120/typescript/004_json_filter/solution.ts
import * as fs from 'fs';
import * as path from 'path';

/**
 * Interface representing the structure of the input data objects.
 */
interface DataRecord {
  name: string;
  age: number;
  active: boolean;
  score: number;
}

/**
 * Main function to process the JSON file.
 * Reads input/data.json, filters by criteria, sorts by name, and outputs JSON.
 */
function processData() {
  // Define the path to the required input file
  const filePath = path.join(process.cwd(), 'input', 'data.json');

  try {
    // Read the file content synchronously
    const fileContent = fs.readFileSync(filePath, 'utf8');
    
    // Parse the JSON array
    const records: DataRecord[] = JSON.parse(fileContent);

    // 1. Filter: active must be true AND age must be 30 or older
    // 2. Sort: name ascending using localeCompare for alphabetical correctness
    const result = records
      .filter((record) => record.active === true && record.age >= 30)
      .sort((a, b) => a.name.localeCompare(b.name));

    // Output the resulting array as a JSON string to stdout
    // We use JSON.stringify without indentation to keep the output clean, 
    // though a single line or formatted JSON are both valid JSON.
    process.stdout.write(JSON.stringify(result) + '\n');
  } catch (error) {
    // The requirement states: "Do not output anything else."
    // Therefore, we catch errors (like file not found or invalid JSON) 
    // and remain silent to ensure only the valid JSON array is output.
  }
}

// Execute the logic
processData();